monkey/monkey_island/cc/services/agent_plugin_service/agent_plugin_service.py

```diff
@@ -30,3 +34,14 @@ def get_all_plugin_configuration_schemas(
 
     def get_all_plugin_manifests(self) -> Dict[AgentPluginType, Dict[str, AgentPluginManifest]]:
         return self._agent_plugin_repository.get_all_plugin_manifests()
+
+    def install_agent_plugin_archive(self, agent_plugin_archive: bytes):
+        with self._lock:
+            os_agent_plugins = parse_plugin(io.BytesIO(agent_plugin_archive))
+            for operating_system, agent_plugin in os_agent_plugins.items():
+                self._agent_plugin_repository.remove_agent_plugin(
+                    None if len(os_agent_plugins) == 1 else operating_system,
```

Comment:
This logic is needlessly complicated and could also lead to an inconsistent system state.

```suggestion
                    None
```